pkg/pool/pool.go

```diff
@@ -715,6 +721,51 @@ func (p *pool) getPDDeviation() int {
 	return deviationPrefix * NumIPv4AddrPerPrefix
 }
 
+// calculateSecondaryIPDeviation calculates the deviation required to meet the desired state for secondary IP mode
+// Returns a number of IPv4 addresses by taking into account the MinIPTarget and WarmIPTarget
+func (p *pool) calculateSecondaryIPDeviation() int {
```

Comment:
To be honest, I am not convinced yet this is thread safe. This function is invoked from a queued work which is executed by async workers.